lucene/sandbox/src/java/org/apache/lucene/document/FloatPointNearestNeighbor.java

```diff
@@ -306,44 +250,36 @@ public String toString() {
       hits[downTo] = hitQueue.poll();
       downTo--;
     }
+    //System.out.println(visitor.comp);
     return hits;
   }
 
-  private static double approxBestDistanceSquared(byte[] minPackedValue, byte[] maxPackedValue, float[] value) {
-    boolean insideCell = true;
-    float[] min = new float[value.length];
-    float[] max = new float[value.length];
-    double[] closest = new double[value.length];
+  private static double pointToRectangleDistanceSquared(byte[] minPackedValue, byte[] maxPackedValue, float[] value, double worstNearestDistanceSquare) {
+    double sumOfSquaredDiffs = 0.0d;
     for (int i = 0, offset = 0 ; i < value.length ; ++i, offset += Float.BYTES) {
-      min[i] = FloatPoint.decodeDimension(minPackedValue, offset);
-      max[i] = FloatPoint.decodeDimension(maxPackedValue, offset);
-      if (insideCell) {
-        if (value[i] < min[i] || value[i] > max[i]) {
-          insideCell = false;
+      double min = FloatPoint.decodeDimension(minPackedValue, offset);
+      if (value[i] < min) {
+        double diff = min - (double)value[i];
+        sumOfSquaredDiffs += diff * diff;
+        if (worstNearestDistanceSquare < sumOfSquaredDiffs) {
```

Comment:
I doubt this shortcut actually helps, does it?

Reply:
Not really, it does not make any difference. I remove it for clarity.